core/modules/replica/ConfigurationIFace.h

```diff
@@ -0,0 +1,1140 @@
+     *   which are not covered by this technology-neutral interface.
+     * @note
+     *
+     * @param name
+     *   the name of a worker
+     *
+     * @param tmpDir
+     *   the new file system path
+     *
+     * @return
+     *   updated worker descriptor
+     *
+     * @throw std::invalid_argument
+     *   if the specified worker was not found in the configuration.
+     */
+    virtual WorkerInfo setWorkerLoaderTmpDir(std::string const& name,
+                                             std::string const& tmpDir) = 0;
+
+    /// @return the name of the default technology for implementing requests
+    virtual std::string workerTechnology() const = 0;
+
+    /// @param val  the new value of the parameter
+    virtual void setWorkerTechnology(std::string const& val) = 0;
+
+    /// @return the number of request processing threads in each worker service
+    virtual size_t workerNumProcessingThreads() const = 0;
+
+    /// @param val  the new value of the parameter
+    virtual void setWorkerNumProcessingThreads(size_t val) = 0;
+
+    /// @return the number of request processing threads in each worker's file service
+    virtual size_t fsNumProcessingThreads() const = 0;
+
+    /// @param val  the new value of the parameter
+    virtual void setFsNumProcessingThreads(size_t val) = 0;
+
+    /// @return the buffer size for the file I/O operations
+    virtual size_t workerFsBufferSizeBytes() const = 0;
+
+    /// @param val  the new value of the parameter
+    virtual void setWorkerFsBufferSizeBytes(size_t val) = 0;
+
+    /// @return the number of request processing threads in each worker's Ingest service
+    virtual size_t loaderNumProcessingThreads() const = 0;
+
+    /// @param val  the new value of the parameter
+    virtual void setLoaderNumProcessingThreads(size_t val) = 0;
+
+    // -----------
+    // -- Misc. --
+    // -----------
+
+    /**
+     * Serialize the configuration parameters into a string
+     *
+     * @return
+     *    string representation of the cached Configuration
+     */
+    virtual std::string asString() const = 0;
+
+    /**
+     * Serialize the configuration parameters into the Logger
+     */
+    virtual void dumpIntoLogger() const = 0;
+
+protected:
+
+    ConfigurationIFace() = default;
+
+    /**
+     * @param func
+     *   (optional) the name of a method/function requested the context string
+     *
+     * @return
+     *   the context string for debugging and diagnostic printouts
+     */
+    std::string context(std::string const& func=std::string()) const;
+
+private:
+
+    // Global parameters of the database connectors (read-write)
+
+    static bool         _databaseAllowReconnect;
+    static unsigned int _databaseConnectTimeoutSec;
+    static unsigned int _databaseMaxReconnects;
+    static unsigned int _databaseTransactionTimeoutSec;
+    static std::string  _qservMasterDatabasePassword;
+    static std::string  _qservWorkerDatabasePassword;
+};
```

Comment:
All `static` parameters (methods and members) have been moved into the public class `Configuration` since this is the only class seen by users.

The sheer number of methods should not be a big problem here because the Configuration service represents a "bag" of mostly independent (of each other) parameters of the system. The current implementation of the service is rather straightforward. So, besides the general aesthetics (where I would agree with your criticism), it shouldn't be a problem for users of the class. The real problem is the amount of the boilerplate code to be added when adding new parameters. I may have a good solution here which would be to group related parameters into structure and have methods to return or set objects of those structures. Each parameter would be a member of the corresponding structure. This could be done (probably) to the majority of the parameters. And, by the way, this approach has already been implemented in the Configuration service with structures `WorkerInfo`, `DatabaseInfo`, `DatabaseFamilyInfo`, etc. I will create a separate "refactoring" ticket for that.

And regarding `setXxx()` methods, I think the Configuration service needs to be symmetric to allow the "two-way" operations with managed parameters.

And thank you for all comments and suggestions! They all, as always, were very useful!

Reply:
Forgot to mention that the problem with the current design of the Configuration service was already recognized. A few months ago there was an improvement registered for that:
https://confluence.lsstcorp.org/display/DM/Refactor+Configuration+service